db/commitlog/commitlog.cc

```diff
@@ -740,8 +749,16 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
     friend std::ostream& operator<<(std::ostream&, const segment&);
     friend class segment_manager;
 
+    size_t sector_overhead(size_t size) const {
+        return (size / (_alignment - detail::sector_overhead_size)) * detail::sector_overhead_size;
+    }
+
     size_t buffer_position() const {
-        return _buffer.size_bytes() - _buffer_ostream.size();
+        // need some arithmetics to figure out what out actual position is, including
+        // page checksums etc. The ostream does not include this.
+        auto size = _buffer_ostream.size();
+        size += sector_overhead(size);
+        return _buffer.size_bytes() - size;
```

Comment:
I don't understand this, especially when there's no documentation about what this function is supposed to return.
If the buffer holds only payload bytes and `_buffer_ostream.size()` counts only payload bytes, why do we need to take into account the `sector_overhead`> and substract it from `_buffer.size_bytes()`?

Reply:
Historical reasons. Actual file position in segment (where we are adding data) is _file_pos (pos to where we've flushed buffers) + position in our memory buffer. 
I.e. the actual buffer position needs to add back the sector overhead.